src/app/js/fields/FieldInternalIcon.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import HomeIcon from '@material-ui/icons/Home';
+import DescriptionIcon from '@material-ui/icons/Description';
+import BarChartIcon from '@material-ui/icons/BarChart';
+
+const HOME = 'home';
+const DOCUMENT = 'document';
+const CHART = 'chart';
+const INTERNAL_STATE_ICON = [];
+INTERNAL_STATE_ICON['home'] = <HomeIcon />;
+INTERNAL_STATE_ICON['document'] = <DescriptionIcon />;
+INTERNAL_STATE_ICON['chart'] = <BarChartIcon />;
+
+const getIconInternalScope = state => {
+    if (!!state) {
+        return INTERNAL_STATE_ICON[state];
+    }
+};
+
+const FieldInternalIcon = ({ state }) => (getIconInternalScope(state));
```

Comment:
Is it still in use?

Reply:
It is still in use,  it is the component which returns the good icon in the selection of the fields and in the datagrid. but the term state doesn't make sense anymore, I just changed it to scope